sw/host/tests/crypto/ecdsa_kat/src/main.rs

```diff
@@ -240,12 +310,147 @@ fn run_ecdsa_testcase(
     }
     .send(&*uart)?;
 
-    let ecdsa_output = CryptotestEcdsaVerifyOutput::recv(&*uart, opts.timeout, false)?;
-    let success = match ecdsa_output {
-        CryptotestEcdsaVerifyOutput::Success => true,
-        CryptotestEcdsaVerifyOutput::Failure => false,
-        CryptotestEcdsaVerifyOutput::IntValue(i) => panic!("Invalid ECDSA verify result: {}", i),
-    };
+    CryptotestEcdsaPrivateKey {
+        d0: ArrayVec::try_from(d0.as_slice()).unwrap(),
+        d0_len: d0.len(),
+        d1: ArrayVec::try_from(d1.as_slice()).unwrap(),
+        d1_len: d1.len(),
+        unmasked_len: d0.len(),
+    }
+    .send(&*uart)?;
+    let success =
+        match operation {
+            CryptotestEcdsaOperation::Sign => {
```

Comment:
The `Sign` arm of this match statement is getting quite long and the nesting is pretty deep. Can this be split out into its own function to improve readability?